packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -801,10 +763,21 @@ class Viewport implements IViewport {
       vtkCamera.setViewAngle(viewAngle);
     }
 
-    if (slabThickness !== undefined && 'setSlabThickness' in vtkCamera) {
-      vtkCamera.setSlabThickness(slabThickness);
-    }
+    // update clippingPlanes
+    this.updateActorsClippingPlanesOnCameraModified(updatedCamera);
 
+    this.triggerCameraModifiedEvent(previousCamera, updatedCamera);
+  }
+
+  /**
+   * Trigger camera modified event
+   * @param cameraInterface - ICamera
+   * @param cameraInterface - ICamera
+   */
+  public triggerCameraModifiedEvent(
```

Comment:
The name of this function implies that it always triggers, which is not the case in its logic